src/main/java/frc/robot/subsystems/CDSSubsystem.java

```diff
@@ -302,67 +380,66 @@ public ManagementState getState() {
   public void changeState() {
     getSensorStatus();
     ballCount = getBallCount();
-    String sensedBallColor = senseColor();
-    int currentOpenSensor = getNextOpenSensor();
-
-    boolean ballPresent =
-        activationArray[2]; // whether or not there's a ball at the centering wheels
+    String[] sensedBallColors = senseAllColors();
+
+    String[] ballLayoutArray = new String[] {"0", "0", "0"};
+    for (int i = 0; i < 3; i++) {
+      if (activationArray[i] && sensedBallColors[i] == allianceColor) {
+        ballLayoutArray[i] = "2";
+      } else if (activationArray[i] && sensedBallColors[i] != allianceColor) {
+        ballLayoutArray[i] = "1";
+      }
+    }
 
-    switch (state) {
-      case IDLE:
-        nextOpenSensor = -1;
-        msCurrent = 0;
+    ballLayout =
+        String.format("%s%s%s", ballLayoutArray[0], ballLayoutArray[1], ballLayoutArray[2]);
```

Comment:
```suggestion
    ballLayout = ballLayoutArray[0] + ballLayoutArray[1] + ballLayoutArray[2];
```

Also, make sure the order is correct. I thought index 2 was the front sensor?